commit 93e2235cb263fc61d46f692d4c237518bec5d38b
Author: bertin <macportsuser@macports.org>
Date:   Fri Oct 10 19:10:41 2025 +0200

    /opt/local/linux-ports/audio/audacity/files/patch-call-XInitThreads.diff

diff --git a/src/AudacityApp.cpp b/src/AudacityApp.cpp
index 4f3ecc990..ddbb5f24b 100644
--- a/src/AudacityApp.cpp
+++ b/src/AudacityApp.cpp
@@ -875,6 +875,7 @@ int main(int argc, char *argv[])
 #elif defined(__WXGTK__) && defined(NDEBUG)
 
 // Specially define main() for Linux debug
+#include <X11/Xlib.h>
 
 IMPLEMENT_APP_NO_MAIN(AudacityApp)
 IMPLEMENT_WX_THEME_SUPPORT
@@ -891,15 +892,17 @@ int main(int argc, char *argv[])
    // never be able to get rid of the messages entirely, but we should
    // look into what's causing them, so allow them to show in Debug
    // builds.
-   freopen("/dev/null", "w", stdout);
-   freopen("/dev/null", "w", stderr);
+//    freopen("/dev/null", "w", stdout);
+//    freopen("/dev/null", "w", stderr);
 
+   XInitThreads();
    return wxEntry(argc, argv);
 }
 
 #elif defined(__WXGTK__)
 
 // Linux release build
+#include <X11/Xlib.h>
 
 wxIMPLEMENT_WX_THEME_SUPPORT
 int main(int argc, char *argv[])
@@ -909,6 +912,7 @@ int main(int argc, char *argv[])
 
    wxDISABLE_DEBUG_SUPPORT();
 
+   XInitThreads();
    return wxEntry(argc, argv);
 }
 wxIMPLEMENT_APP_NO_MAIN(AudacityApp);
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 909028874..5b1c14b5c 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1018,6 +1018,7 @@ set( LIBRARIES
       $<$<BOOL:${USE_VAMP}>:libvamp>
       $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:PkgConfig::GLIB>
       $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:PkgConfig::GTK>
+      $<$<PLATFORM_ID:Linux,FreeBSD,OpenBSD,NetBSD,CYGWIN>:-lX11>
       $<$<TARGET_EXISTS:Threads::Threads>:Threads::Threads>
 )
 
